src/Microsoft.ML.Transforms/OneHotEncoding.cs

```diff
@@ -177,37 +150,64 @@ public sealed class OneHotEncodingEstimator : IEstimator<OneHotEncodingTransform
+            Binary = 4,
         }
 
         /// <summary>
         /// Describes how the transformer handles one column pair.
         /// </summary>
         public sealed class ColumnOptions : ValueToKeyMappingEstimator.ColumnOptionsBase
         {
-            public readonly OneHotEncodingTransformer.OutputKind OutputKind;
+            public readonly OutputKind OutputKind;
             /// <summary>
             /// Describes how the transformer handles one column pair.
             /// </summary>
             /// <param name="name">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
             /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="name"/> will be used as source.</param>
             /// <param name="outputKind">Output kind: Bag (multi-set vector), Ind (indicator vector), Key (index), or Binary encoded indicator vector.</param>
-            /// <param name="maxNumTerms">Maximum number of terms to keep per column when auto-training.</param>
-            /// <param name="sort">How items should be ordered when vectorized. If <see cref="ValueToKeyMappingEstimator.SortOrder.Occurrence"/> choosen they will be in the order encountered.
-            /// If <see cref="ValueToKeyMappingEstimator.SortOrder.Value"/>, items are sorted according to their default comparison, for example, text sorting will be case sensitive (for example, 'A' then 'Z' then 'a').</param>
-            /// <param name="term">List of terms.</param>
+            /// <param name="maxNumberOfKeys">Maximum number of terms to keep per column when auto-training.</param>
+            /// <param name="mappingOrder">How items should be ordered when vectorized. If <see cref="ValueToKeyMappingEstimator.MappingOrder.ByOccurrence"/> choosen they will be in the order encountered.
+            /// If <see cref="ValueToKeyMappingEstimator.MappingOrder.ByValue"/>, items are sorted according to their default comparison, for example, text sorting will be case sensitive (for example, 'A' then 'Z' then 'a').</param>
+            /// <param name="keys">List of terms.</param>
             public ColumnOptions(string name, string inputColumnName = null,
-                OneHotEncodingTransformer.OutputKind outputKind = Defaults.OutKind,
-                int maxNumTerms = ValueToKeyMappingEstimator.Defaults.MaxNumKeys, ValueToKeyMappingEstimator.SortOrder sort = ValueToKeyMappingEstimator.Defaults.Sort,
-                string[] term = null)
-                : base(name, inputColumnName ?? name, maxNumTerms, sort, term, true)
+                OutputKind outputKind = Defaults.OutKind,
+                int maxNumberOfKeys = ValueToKeyMappingEstimator.Defaults.MaximumNumberOfKeys, ValueToKeyMappingEstimator.MappingOrder mappingOrder = ValueToKeyMappingEstimator.Defaults.Order,
+                string[] keys = null)
```

Comment:
>string[] keys = null [](start = 16, length = 20)

I don't understand why we expose it.
I can be wrong, but I had feeling what we can apply OneHotEncoding not only for strings, but also for floats and pretty much everything.
Why keys are strings? It make sense in command line world, you couldn't pass anything else.
But right now, I don't get.
I would understand if we could take array of any elements and create KeyType out of it as object, and then pass it here, that would be cool.
But this one, I just don't get it. I would rather hide it from user.

Reply:
I agree, I am removing it from all the transforms where we were exposing it. We always have an alternative and it does not make much sense to keep using this format. 

---
In reply to: [263642179](https://github.com/dotnet/machinelearning/pull/2862#discussion_r263642179) [](ancestors = 263642179)